### PR TITLE
[frontend] navbar distinguish roles

### LINKED ISSUE #132: [frontend] pages need to deal with 401, 403 with user access
**Describe the bug**
different types of user can access to different pages, some of which should block.
currently admin can't access seller section, when click the section, it will give us `401` or `403`
1. use user token to prevent user click into a page that backend is not available while frontend gives it wrong page view
2. the `navbar` should update to check user from token to catalog user

**To Reproduce**
after #113 merged, it will appears

**Screenshots**
### backend is not available while frontend gives it wrong page view example
it should be /unauthorized view
image
### access to not valid page should looks like this
image

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by KatoToyoki**
Resolveeeeeeeees part of #94

### admin
- login with admin, the navbar looks like
    - image
    - image
    - image
    - image

### customer
- login with customer, the navbar looks like
    - image
    - image
    - image
    - image

### Test Environment Setup
open 3 terminals
one for `docker-compose up postgres minio `
one for `go run .`
one for `cd frontend` -> `npm i` -> `npm run build`
then open the browser then enter http://localhost:8080/ 

### Additional context

N/A

### CODE REVIEW

**Comment by UGisBusy** [comment]
image
in Admin pov, these options are needed?
this might need to Beckend check
also, would be better if admin text is on the left side

File: frontend/src/components/NavBar.tsx [hunk 1]
```diff
@@ -70,52 +69,56 @@ const NavBar = () => {
...
+                            className='none nav_link'
+                            style={{ padding: '0%' }}
+                          >
+                            <div style={{ padding: '5px 10% 5px 10%' }}>All Reports</div>
+                          </Link>
+                        </Dropdown.Menu>
+                      </Dropdown>
+                    ) : (
+                      ''
+                    )}
```
Comment (NOOBDY) [inline frontend/src/components/NavBar.tsx]:
```suggestion
                    ) : null}
```

Comment (NOOBDY) [reply frontend/src/components/NavBar.tsx]: you can also change to this somewhere down this component

**Review by NOOBDY — CHANGES_REQUESTED** [review]

**Comment by x06lan** [comment]
> image in Admin pov, these options are needed? this might need to Beckend check also, would be better if admin text is on the left side

the  security and order_history can remove but the personal info tab need to stay for change password admin image etc...

**Comment by KatoToyoki** [comment]
> image in Admin pov, these options are needed? this might need to Beckend check also, would be better if admin text is on the left side



> > image in Admin pov, these options are needed? this might need to Beckend check also, would be better if admin text is on the left side
> 
> the security and order_history can remove but the personal info tab need to stay for change password admin image etc...

screen shots updated

**Review by Error0229 — DISMISSED** [review]
WOMM

**Review by NOOBDY — DISMISSED** [review]

**Review by NOOBDY — APPROVED** [review]

**Review by Error0229 — APPROVED** [review]

**Pull request merged**